web/src/copy_and_paste.js

```diff
@@ -293,29 +293,138 @@ function get_end_tr_from_endc($endc) {
     return $endc.parents(".selectable_row").first();
 }
 
+function cleanAttribute(attribute) {
+    return attribute ? attribute.replace(/(\n+\s*)+/g, "\n") : "";
+}
+
+function image_to_zulip_markdown(content, node) {
+    if (node.nodeName === "IMG" && node.classList.contains("emoji") && node.hasAttribute("alt")) {
+        // For zulip's custom emojis
+        return node.getAttribute("alt");
+    }
+    const src = node.getAttribute("src") || node.getAttribute("href") || "";
+    const title = cleanAttribute(node.getAttribute("title")) || "";
+    // Using Zulip's link like syntax for images
+    return src ? "[" + title + "](" + src + ")" : node.getAttribute("alt") || "";
+}
+
 export function paste_handler_converter(paste_html) {
-    const turndownService = new TurndownService();
+    // turning off escaping (for now) to remove extra `/`
+    TurndownService.prototype.escape = (string) => string;
+
+    const turndownService = new TurndownService({emDelimiter: "*", codeBlockStyle: "fenced"});
     turndownService.addRule("headings", {
         filter: ["h1", "h2", "h3", "h4", "h5", "h6"],
         replacement(content) {
             return content;
         },
     });
-    turndownService.addRule("emphasis", {
-        filter: ["em", "i"],
+    turndownService.addRule("strikethrough", {
+        filter: ["del", "s", "strike"],
         replacement(content) {
-            return "*" + content + "*";
+            return "~~" + content + "~~";
         },
     });
-    // Checks for raw links without custom text or title.
     turndownService.addRule("links", {
+        filter: ["a"],
+        replacement(content, node) {
+            if (node.href === node.innerHTML) {
+                // Checks for raw links without custom text.
+                return content;
+            }
+            if (node.childNodes.length === 1 && node.firstChild.nodeName === "IMG") {
+                // ignore link's url if it only has an image
+                return content;
+            }
+            return "[" + content + "](" + node.href + ")";
```

Comment:
This probably wants a special case for where the `node.href == content`; in that case, we probably just want to emit `content`. (I.e. if we're copying a URL from another tool that linkifies URLs, like Gmail)

Reply:
The first `if` check is for exactly that, since `content` is essentially the `node`'s `innerHTML`. But yes, I've changed `node.innerHTML` to `content` for consistency